egs2/aishell4/asr1/local/data.sh

```diff
@@ -0,0 +1,359 @@
+        for f in $FILES
+        do
+            echo "$f" >> ${AISHELL4}/$room_name/TextGrid_list.txt
+        done
+
+    done
+fi
+
+
+#################################################################
+#####            Join train_L, train_M and train_S       ########
+#################################################################
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then 
+
+    mkdir -p ${AISHELL4}/full_train
+    for r in train_L train_M train_S ; do 
+        cat ${AISHELL4}/$r/TextGrid_list.txt >> ${AISHELL4}/full_train/TextGrid_list.txt
+        cat ${AISHELL4}/$r/wav_list.txt >> ${AISHELL4}/full_train/wav_list.txt
+    done
+fi
+
+
+
+#################################################################
+#####            ground truth for asr, using aishell4 github     ##############
+#################################################################
+
+
+wav_list_aishell4=${AISHELL4}/full_train/wav_list.txt
+text_grid_aishell4=${AISHELL4}/full_train/TextGrid_list.txt
+
+output_folder=$PWD/data/
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] ; then 
+
+    log "generating asr training data ..."
+    log "(this can take some time)"
+ 
+    python "$FOLDER"/data_preparation/generate_asr_trainingdata.py  --output_dir "$output_folder" --mode train --aishell4_wav_list "$wav_list_aishell4" --textgrid_list "$text_grid_aishell4" || log "ca a pas marché" ;
+
+    log "asr training data generated."
+
+fi
+ 
+
+
+
+#################################################################
+#####     creating wav.scp from output/train/wav directory    ##############
+#################################################################
+
+
+if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ] ; then 
+    FILES="$output_folder/train/wav/*"
+    for f in $FILES
+    do
+        g=$(echo $f | cut -d'/' -f 14 | cut -d'.' -f 1) 
+        echo "$g" "$f" >> $output_folder/train/wav.scp
+    done
+
+fi
+
+
+#################################################################
+#####            creating utt2spk and spk2utt  ########
+#################################################################
+
+if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ] ; then 
+    FILES="$output_folder/train/wav/*"
+    for f in $FILES
+    do
+        g=$(echo $f | cut -d'/' -f 14 | cut -d'.' -f 1) 
+        echo "$g" "$g"  >> $output_folder/train/utt2spk  # we put speaker_id = utt_id
+    done
+
+
+fi 
+
+
+
+
+
+
+
+#################################################################
+#####            sort and fix the data  ########
+#################################################################
+
+
+if [ ${stage} -le 6 ] && [ ${stop_stage} -ge 6 ] ; then
+    log "sorting files ... "
+    sort data/train/utt2spk -o data/train/utt2spk
+    # creating spk2utt from utt2spk
+    utils/utt2spk_to_spk2utt.pl $output_folder/train/utt2spk > $output_folder/train/spk2utt
+    sort data/train/wav.scp -o data/train/wav.scp
+    sort data/train/text -o data/train/text
```

Comment:
Also, I failed to find out where `data/train/text` was generated. Could you give me a pointer if I missed something?
> Similarly for `data/dev/text` and `data/test/text`.

Reply:
Hello, thank you for your comments, I will review those parts very soon and let you know about it. It is true that at some points I modified the code from [felixfuyihui/AISHELL-4](https://github.com/felixfuyihui/AISHELL-4) repo because we only wanted to focus on the ASR part. I will take care of this in the next week

Reply:
Thanks!

Reply:
Hello, I am currently reviewing it, however I'll need bit more time so will try to clearly explain the differences with [felixfuyihui/AISHELL-4](https://github.com/felixfuyihui/AISHELL-4) repo and fix the bugs in the next week, sorry for the delays.

Reply:
> Hello, I am currently reviewing it, however I'll need bit more time so will try to clearly explain the differences with [felixfuyihui/AISHELL-4](https://github.com/felixfuyihui/AISHELL-4) repo and fix the bugs in the next week, sorry for the delays.

Sure. Thank you for taking care of it!